Formula/gap.rb

```diff
@@ -11,46 +18,127 @@ class Gap < Formula
   # So we depend on GNU readline here.
   depends_on "readline"
 
+  # for packages
+  depends_on "cddlib"   # CddInterface
+  depends_on "curl"     # curlInterface
+  depends_on "fplll"    # float
+  depends_on "libmpc"   # float
+  depends_on "mpfi"     # float
+  depends_on "mpfr"     # float, normalizinterface
+  depends_on "ncurses"  # browse
+  depends_on "zeromq"   # ZeroMQInterface
+  depends_on "singular" # many packages
+  depends_on "pari"     # many packages
+
   def install
-    # XXX:  Currently there is no `install` target in `Makefile`.
-    #   According to the manual installation instructions in
-    #
-    #     https://github.com/gap-system/gap/blob/master/INSTALL.md
-    #
-    #   the compiled "bundle" is intended to be used "as is," and there is
-    #   no instructions for how to remove the source and other unnecessary
-    #   files after compilation.  Moreover, the content of the
-    #   subdirectories with special names, such as `bin` and `lib`, is not
-    #   suitable for merging with the content of the corresponding
-    #   subdirectories of `/usr/local`.  The easiest temporary solution seems
-    #   to be to drop the compiled bundle into `<prefix>/libexec` and to
-    #   create a symlink `<prefix>/bin/gap` to the startup script.
-    #   This use of `libexec` seems to contradict Linux Filesystem Hierarchy
-    #   Standard, but is recommended in Homebrew's "Formula Cookbook."
-
-    libexec.install Dir["*"]
-
-    # GAP does not support "make install" so it has to be compiled in place
-
-    cd libexec do
-      system "./configure", "--with-readline=#{Formula["readline"].opt_prefix}"
-      system "make"
+
+
+    no_compilation_packages = [
+      "atlasrep", "aclib", "agt", "alnuth", "automata", "automgrp",
```

Comment:
Such a long list seems error prone and annoying because it needs to be updated for each GAP release.

Can't you instead just compute a list of the names of all subdirs of the `pkg` directory on the fly? Resp. not even use that list, just iterate over them all